src/fdcache.cpp

```diff
@@ -750,10 +750,17 @@ bool FdManager::Close(FdEntity* ent, int fd)
     return false;
 }
 
-bool FdManager::ChangeEntityToTempPath(FdEntity* ent, const char* path)
+bool FdManager::ChangeEntityToTempPath(const FdEntity* ent, const char* path)
 {
     const std::lock_guard<std::mutex> lock(FdManager::except_entmap_lock);
-    except_fent[path] = ent;
+
+    // Search raw FdEntity pointer in map and copy std::shared_ptr.
+    for(fdent_map_t::const_iterator iter = fent.begin(); iter != fent.end(); ++iter){
+        if(iter->second.get() == ent){
+            except_fent[path] = iter->second;   // copy
+        }
+    }
```

Comment:
@ggtakec could you look at #2541?  I believe that using `shared_from_this` will remove the need for this lookup.